timeout reached during docker volumes removal

```shell
+ terraform destroy --auto-approve
random_password.k3s_token: Refreshing state... [id=none]
docker_volume.k3s_server: Refreshing state... [id=k3s-server-master]
docker_network.k3s: Refreshing state... [id=96bfb90dcf4591594f3da7219e3ea15c74b0b86fdee6d933a34e2d9e576ad49d]
docker_image.k3s: Refreshing state... [id=sha256:74d7efe646152231ea0b446c10d99403708f3c7957a0789674c265ed4e4afb7brancher/k3s:v1.19.2-k3s1]
docker_container.k3s_agent[0]: Refreshing state... [id=5a335ece8e3de5ebcdf839fde32f8bff21b107c245d09c9bd4099ca8b6c1a214]
docker_container.k3s_server: Refreshing state... [id=ca224ae0a318ee97b8bfa0dca2f62783086d7d6965963ab7fd87d3800055a4e2]
docker_volume.k3s_server: Destroying... [id=k3s-server-master]
random_password.k3s_token: Destroying... [id=none]
random_password.k3s_token: Destruction complete after 0s
docker_volume.k3s_server: Still destroying... [id=k3s-server-master, 10s elapsed]
docker_volume.k3s_server: Still destroying... [id=k3s-server-master, 20s elapsed]
docker_volume.k3s_server: Still destroying... [id=k3s-server-master, 30s elapsed]

Error: timeout while waiting for state to become 'removed' (last state: 'in_use', timeout: 30s)


make: *** [Makefile:104: clean] Error 1
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -14,3 +14,4 @@ jobs:
       - uses: actions/checkout@v2
       - run: make debug
       - run: make
+      - run: make clean

--- a/Makefile
+++ b/Makefile
@@ -119,6 +119,18 @@ clean: get-base-domain
 		--entrypoint "" \
 		--workdir /workdir \
 		hashicorp/terraform:$(TERRAFORM_VERSION) /workdir/scripts/destroy-vault.sh
+	docker run --rm \
+		--user $(UID_NUMBER):$(GID_NUMBER) \
+		-v $$PWD:/workdir \
+		-v $$PWD/$(ARTIFACTS_DIR)/kubeconfig.yaml:/tmp/.kube/config \
+		--network k3s-$(CLUSTER_NAME) \
+		--env HOME=/tmp \
+		--env KUBECTL_COMMAND=apply \
+		--env ARGOCD_OPTS="--plaintext --port-forward --port-forward-namespace argocd" \
+		--env ARTIFACTS_DIR=$(ARTIFACTS_DIR) \
+		--entrypoint "" \
+		--workdir /workdir \
+		argoproj/argocd:v$(ARGOCD_VERSION) /workdir/scripts/pre-clean.sh
 	touch $$HOME/.terraformrc
 	docker run --rm \
 		--group-add $(DOCKER_GID_NUMBER) \

--- a/scripts/pre-clean.sh
+++ b/scripts/pre-clean.sh
@@ -1,0 +1,21 @@
+#!/bin/sh -xe
+
+# Remove all Applications to prevent ArgoCD to reconcile
+kubectl -n argocd delete application --all
+
+# Remove all Operators to ensure that it does not redeploy stuffs
+kubectl -n cluster-operators delete deployments --all
+
+# Remove all pods to release volume mounted with rshared propagation
+kubectl delete daemonsets --all --all-namespaces
+kubectl delete statefulsets --all --all-namespaces
+kubectl delete deployments --all --all-namespaces
+kubectl delete cronjobs --all --all-namespaces
+kubectl delete jobs --all --all-namespaces
+kubectl delete horizontalpodautoscaler --all --all-namespaces
+kubectl delete service --all --all-namespaces
+
+while test "$(kubectl get pods --all-namespaces | wc -l)" -ne 0; do
+	echo Waiting for destruction of all pods
+	sleep 3
+done

